### ISSUE TITLE
Support docker-compose without Sail

### ISSUE THREAD

**Issue by dorkster100** [adalessa/laravel.nvim#7]
The project I work on does not use Laravel Sail, but I do use docker-compose. It would be nice to make the plugin extendable and usable without Sail. Do I understand correctly that we only need to add a setting to replace `/vendor/bin/sail` and have `/vendor/bin/sail` as default?

Usage i have in mind is to use `docker-compose exec app ` to replace `/vendor/bin/sail`

**Comment by adalessa** [comment]
I've been thinking about that. The problem is that to run artisan commands and other from the container needs to know more about the container, or the expectation. change that will work for some commands but not all of them and will break the sail command that expects to run as the docker compose directly. 
So I don't think about replacing sail but maybe adding configuration for a new "context" or something similar.

**Comment by dorkster100** [comment]
I have looked into the sail bash script, it is quite simple and acts mostly as a proxy for different types of commands, like artisan is sent to docker-compose exec app...
It also gets all the container info from .env file.

Another issue that I ran into trying to introduce new setting, is that actually plenary nvim Job:New() expects an executable, so passing 'docker-compose exec app' does not work.

For now I had success with replacing sail executable with a slightly simpler version of it and using that. One of the things I had to remove for instance was '-u sail' on every command.

In general I also see as a simple solution, to simply expose custom executor for every type of command like
```
npm_exec = "docker-compose exec app"
db_exec = "docker-compose exec db"
```

And add these as settings having the /vendor/bin/sail as default.

The only issue to figure out here would be the job:new I mentioned earlier.

I will look into it more tomorrow.


One other thing, does :Laravel routes and then selecting routes work for you? I fixed it in my setup, I can submit PR

**Comment by adalessa** [comment]
Yes the sail script does a lot of proxy and takes a lot of possible arguments or forward to the app.
The problem with the job is that the job expects a table so passing `docker-compose exec` will break needs to be a table split by the spaces.

I will tink about the possiblility of defining the executables for each one. The db if you do it through artisan command should work no need to go through docker.

Related to the routes yes, it works for me. If you found a bug in your use case please report and I more than happy to review a PR if improve the code to support more use cases.

**Comment by adalessa** [comment]
implemented solution on #15
